File: java-midterm/src/main/java/com/pht/service/BrandService.java
package com.pht.service;

import java.util.List;

import com.pht.dto.request.BrandRequest;
import com.pht.entity.BrandEntity;

public interface BrandService {
	List<BrandEntity> getAll();

	BrandEntity create(BrandRequest brandRequest);

	BrandEntity update(BrandRequest brandRequest, long id);

	void delete(long id);
}

File: midterm/src/app/services/fetch/brand.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './base.service';
import { Observable } from 'rxjs';
import { BrandResponse } from '../../models/brand';

@Injectable({
  providedIn: 'root'
})
export class BrandService {
  constructor(private readonly baseService: BaseService) { }

  getAll(): Observable<BrandResponse[]> {
    return this.baseService.get<BrandResponse[]>('/api/brands/');
  }

  create(formData: FormData): Observable<BrandResponse> {
    return this.baseService.post<BrandResponse>('/api/brands/', formData, true);
  }

  update(id: number, formData: FormData): Observable<BrandResponse> {
    return this.baseService.put<BrandResponse>(`/api/brands/${id}/`, formData, true);
  }

  delete(id: number): Observable<any> {
    return this.baseService.delete<any>(`/api/brands/${id}/`);
  }
}
